src/app/address/address.component.html

```diff
@@ -0,0 +1,65 @@
+
+  <div class="container">
+    <div style="margin-top: 50px; margin-left: 20%; margin-right: 20%;">
+      <div class="panel panel-default">
+        <div class="panel-heading"> <h3 style="text-align:center;">Address Form</h3></div>
+        <div class="panel-body">
+      <div class="row">
+          <div class="col-md-8 offset-md-2">
+
+              <form name="form" (ngSubmit)="addressform.form.valid && onSubmit()" #addressform="ngForm" novalidate>
+                  <div class="form-group">
+                      <label for="streetName"> Street</label>
+                      <input type="text" class="form-control input-lg" name="streetName" [(ngModel)]="model.streetName" #streetName="ngModel" [ngClass]="{ 'is-invalid':  streetName.invalid }" required pattern="\d{1,3}.?\d{0,3}\s[a-zA-Z]{2,30}\s[a-zA-Z]{2,15}" />
+                      <div *ngIf="streetName.invalid" class="invalid-feedback">
+                          <div *ngIf="streetName.errors.required">Street name is required.</div>
+                          <div *ngIf="streetName.errors.pattern">Street name is not in correct format<br> example: 16 bontanic avenue</div>
+                      </div>
+                  </div>
+                  <div class="form-group">
+                      <label for="suburb">Suburb</label>
+                      <input type="text" class="form-control" name="suburb" [(ngModel)]="model.suburb" #suburb="ngModel" [ngClass]="{ 'is-invalid':suburb.invalid }" required appTextOnly />
+                      <span class="invalid-feedback">
+                          <span *ngIf="suburb.errors?.required">suburb is required</span>
+                          <span *ngIf="suburb.errors?.textvalidation">no numbers allowed ,letters only.</span>
+                      </span>
+                  </div>
+                  <div class="form-group">
+                    <label for="city">City</label>
+                    <input type="text" class="form-control" name="city" [(ngModel)]="model.city" #city="ngModel" [ngClass]="{ 'is-invalid':city.invalid }" required appTextOnly />
+                    <div *ngIf="city.invalid" class="invalid-feedback">
+                        <div *ngIf="city.errors.required">City is required</div>
+                        <span *ngIf="city.errors?.textvalidation">no numbers allowed ,letters only</span>
+                    </div>
+                </div>
+
+                <div class="form-group">
+                  <label for="province">province</label>
```

Comment:
Please apply a consistent naming convention for your labels. I think this should be "Province" rather and not "province"

Reply:
@strizzwald fixed